File: EmployeeDirectoryApp/src/app/models/employee.ts
export class Employee {
  id?: string = '';
  department: string = '';
  email: string = '';
  firstName: string = '';
  image?: string;
  lastName: string = '';
  office: string = '';
  phoneNumber: string = '';
  skypeId: string = '';
  jobTitle: string = '';
}
export enum FilterOption {
  FirstName = 'firstName',
  LastName = 'lastName',
  Email = 'email',
  Department = 'department',
  Office = 'office',
}

export class User {
  name!: string;
  role:string = "user";
  email!: string;
  username!: string;
  password!: string;
}

export class EmployeeFilterCriteria {
  Department? :string ="";
   Office?:string ="";
   JobTitle?:string =""
}

File: EmployeeManagerAPI/EmployeeManagerAPI/Models/Employee.cs
using EmployeeManagerAPI.Models.DTO;

namespace EmployeeManagerAPI.Models
{
    public class Employee
    {
        public string? Id { get; set; }
        public string? Department { get; set; }
        public string? Email { get; set; }
        public string? FirstName { get; set; }
        public string? Image { get; set; }
        public string? LastName { get; set; }
        public string? Office { get; set; }
        public string? PhoneNumber { get; set; }
        public string? SkypeId { get; set; }
        public string? JobTitle { get; set; }

        public static implicit operator Employee(EmployeeDto v)
        {
            throw new NotImplementedException();
        }
    }

}
